### PR TITLE
SLING-12027 - GraphQL core fails to build with Java 17

### PR DESCRIPTION
**Pull request by rombert**

### CODE REVIEW

**Comment by rombert** [comment]
@raducotescu @schaefa - I think this is mostly done, but one test fails and it requires more GraphQL insight

> [ERROR] Failures: 
> [ERROR]   GraphQLServletIT.testPersistingInvalidQueries:159 Did not expect to persist an invalid query. expected:<400> but was:<500>

It would be great if you could look into this so we can have Java 17 compatibility for this bundle.

**Comment by schaefa** [comment]
@rombert I will check it out

**Comment by raducotescu** [comment]
@schaefa, the code fails on Java 11 in the same way.

File: pom.xml [hunk 1]
```diff
@@ -249,7 +249,7 @@
     <dependency>
       <groupId>org.apache.sling</groupId>
       <artifactId>org.apache.sling.testing.paxexam</artifactId>
-      <version>3.1.0</version>
+      <version>4.0.0</version>
```
Comment (raducotescu) [inline pom.xml:252]: This seems to be the culprit of the failing test.

Comment (rombert) [reply pom.xml:252]: I added this for Java 17 support, see https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=255071690#CommonIssueswhenbuildingwithnewerJDKs(11or17)-Pax-Examtestfailures

Comment (raducotescu) [reply pom.xml:252]: This update brought `org.apache.felix.jetty-5.0.0`, which uses Jetty 11. Unfortunately, there's a Servlet Request Wrapper in Felix that doesn't correctly return the set attributes if they were `RequestDispatcher` attributes - see https://github.com/apache/felix-dev/blob/d691a59c9e78749fad20a10a002c969e4e583800/http/wrappers/src/main/java/org/apache/felix/http/javaxwrappers/ServletRequestWrapper.java#L108-L182

@cziegeler will fix this in Felix and for this test setup we could override the Jetty version.

Comment (rombert) [reply pom.xml:252]: Nice catch!

**Comment by rombert** [comment]
Thanks @schaefa , @raducotescu , @cziegeler for your help, this has been a fun ride :-)

**Pull request merged**